pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -481,6 +481,8 @@ func (is *InstanceService) InstanceCreate(clusterName string, name string, confi
 		UserData:         []byte(userData),
 		SecurityGroups:   securityGroups,
 		ServiceClient:    is.computeClient,
+		Tags:             config.Tags,
```

Comment:
I was thinking about this a bit more and I think we should add the machine tags to all the resources that are created as part of the instance creation. Therefore, the ports created in this function should also have `config.Tags` added to them.

Also, what happens if you run this code against a Nova that doesn't have support for Tags? Can you check what OpenStack release added tags to Nova?

Reply:
Yeah, I think it would makes sense to create a "Global Tags" field of tags that get applied to everything created by CAPO

Reply:
https://developer.openstack.org/api-ref/compute/#server-tags-servers-tags 
2.26 added tag

Reply:
mmh, gophercloud says we should be using 2.52 :/

Reply:
ok. 2.26 means support add tags and 2.52 means support add tags *during* boot command
 :)